doc/nyan/base types.md

```diff
@@ -0,0 +1,164 @@
+	# after recharge_time in seconds.
+	recharge_time : float
+
+GatherAbility(Ability): # WIP
+	# Gathering resources requires this Ability
+	# WIP1: do we need a member for the current amount and type of resource here? or is that only inside the engine?
+
+	allowedResource : set(Resource) # Allowed ressources?
+	maxAmount : int # maximum amount  that can be carried
+	speed : float # gathered units per second
+
+	# WIP2: proposal: allow different maximum amounts for different resources:
+
+#	allowedResource : orderedset(Resource)
+#	maxAmount : orderedSet(int) # does <10,5,0,10> work? it is a set after all.
+
+	# WIP3: if WIP2 is not possible: implement an 'Ability' that allows to gather only one resource.
+	# A worker would then need 4 gather abilities added to it's 'abilities' set.
+	# And this would allow to reference
+
+	# WIP4: how do we reference Animations?
+#	animation : file # one picture that contains all frames or, if a directory,
+	                # contains all frames. # WIP5: naming scheme?
+#	renameMe : file # WIP6: how to read `animation'? describes, where in the big picture `animation' which frames are; whether to potentially mirror them.
+	                # possibilities: 1. $renameMe is a regular file: description of `animation' (syntax?)
+	                # 2. replace `renameMe' by more fields that describe `animation' (ugly imo, but maybe the best way)
+	                # 3. remove `renameMe':
+	                #    if $animation is a directory: look for '$animation/animate.nfo'
+	                #    else: look for "$(basename $animation png)nfo"
+
+CarryAbility(Ability): # WIP: find a better name?
+	# Used by priests for relicts.
+	# WIP1: how should this look like? keep it general? can maybe be used by buildings to let units find shelter in them? or by mods for adding new relict-like objects
+	# WIP2: can maybe merged with a modified GatherAbility?
+
+#	goodMemberName : Portable # what can be collected? make a 'Portable' class and let relict inherit that? could then be also used by resources?
+#	maxAmount
+
+	animation : file # what does a priest look like when he's carrying it?
+	                 # WIP: see GatherAbility.animation and nyan question 4 (further down).
+
+AttackAbility(Ability):
+	# Every attack shall inherit from this.
+	# WIP: how will damage calculation work in oa?
+
+	range : float
+	rate : float
+	damage : float # rename to baseDamage ?
+
+
+Unit(): # WIP
+	# Everything you can see in the game # (WIP: but not assets, which is basically everything you see. Maybe better:)
+	# Everything that has some 'Ability' inherits from this.
+
+	ability : set(Ability)
+
+
+
+#and probably more. if you know what's missing, add it.
+
+```
+
+Open nyan implementation questions:
+
+1. How do we say in Nyan, that a Towncenter attack depends on how many people are inside the building?
+2. What about the building 'Ability' to carry units? will this lead to recursive references?
```

Comment:
Maybe we can define types of units: infantry, cavalry, ships, siege. this could help with the allowed_targets member in attackability and with the upgrades from research.